modules/configuration/src/main/java/org/apache/ignite/internal/configuration/ConfigurationRegistry.java

```diff
@@ -96,8 +83,11 @@ public class ConfigurationRegistry implements IgniteComponent, ConfigurationStor
     /** Configuration change handler. */
     private final ConfigurationChanger changer;
 
-    /** Configuration generator. */
-    private final ConfigurationAsmGenerator cgen = new ConfigurationAsmGenerator();
+    /** Runtime implementations generator for node classes. */
+    private final ConfigurationTreeGenerator generator;
+
+    /** Flag that indicates if the {@link ConfigurationTreeGenerator} instance is owned by this object or not. */
+    private boolean ownConfigTreeGenerator = false;
```

Comment:
Why does it matter?
I believe that generator can always be a constructor parameter, and another component, that instantiated it, should also close it in the end

Reply:
I don't think I understand the answer. Let me rephrase myself.
Why do you need a flag "ownConfigTreeGenerator"? It can be removed, right?
Second question - why do you need a "close" method in the generator? I don't see a reason to have it, other than writing `cgen.close()` instead of `cgen = null` in some tests. Can you please explain your decision?

Reply:
Technically you are right, we might write `cgen = null`. But this is something hidden. How developers might know that they should nullify the link because we don't want them to hold the link and produce memory leaks? 

I decided to move this contract into the place where contracts are defined – the interface. I think a clear declaration of `close` method helps developers to understand how to work with the object.